### PR TITLE
Make decision of new units by relying on new tests dir size.

### PR DESCRIPTION
**Pull request by inferno-chromium**
new_units_added metric is not always available, esp in fork mode.

### CODE REVIEW

File: src/python/bot/fuzzers/libFuzzer/launcher.py [hunk 1]
```diff
@@ -980,7 +980,9 @@ def main(argv):
 
   # Make a decision on whether merge step is needed at all. If there are no
   # new units added by libFuzzer run, then no need to do merge at all.
-  new_units_added = parsed_stats.get('new_units_added', 0)
+  new_units_added = (
+      shell.get_directory_file_count(new_testcases_directory) or
+      parsed_stats.get('new_units_added'))  # redundant, but needed for tests.
```
Comment (jonathanmetzman) [inline src/python/bot/fuzzers/libFuzzer/launcher.py]: Can you change the comment to a TODO to mock `shell.get_directory_file_count`. I think this is fine as a temporary solution but seems weird to have long term unless absolutely necessary.

**Comment by jonathanmetzman** [comment]
LGTM

File: src/python/bot/fuzzers/libFuzzer/launcher.py (hunk 1 above)
Comment (oliverchang) [reply src/python/bot/fuzzers/libFuzzer/launcher.py]: +1. mocking is definitely better here.

Comment (inferno-chromium) [reply src/python/bot/fuzzers/libFuzzer/launcher.py]: fixed.
minijail test was broken since self.command[0] is not _fuzzer there but minijail bin path.
also test_oom test ends up showing case where merge units=0, so we have a way to emulate that using None in mock_popen, so used that.

File: src/python/tests/core/bot/fuzzers/libFuzzer/libfuzzer_launcher_test.py [hunk 2]
```diff
@@ -1190,8 +1190,11 @@ def test_oom_crash(self, mock_stdout):
     self.fs.CreateDirectory('/fake/corpus_oom')
     os.environ['FUZZ_CORPUS_DIR'] = '/fake/corpus_oom'
 
-    with mock.patch('subprocess.Popen',
-                    create_mock_popen(self.oom_output)) as mock_popen:
+    new_units_added = 11
+    with mock.patch(
+        'subprocess.Popen',
+        create_mock_popen(self.oom_output, '/fake/inputs-disk/temp-1337/new',
+                          None, new_units_added)) as mock_popen:
```
Comment (inferno-chromium) [inline src/python/tests/core/bot/fuzzers/libFuzzer/libfuzzer_launcher_test.py]: None is used to show

              'new_units_added':
                  0,	                  0,

which just means we tried merge (since new_testcase_directory has units), but then running with -merge=1 failed, so units=0.

File: src/python/tests/core/bot/fuzzers/libFuzzer/libfuzzer_launcher_test.py [hunk 3]
```diff
@@ -100,7 +100,7 @@ def _write_fake_units(self):
 
     def communicate(self, input_data=None):
       """Mock subprocess.Popen.communicate."""
-      if self.command[0].endswith('_fuzzer'):
+      if '/fake/build_dir/fake_fuzzer' in self.command:
```
Comment (inferno-chromium) [inline src/python/tests/core/bot/fuzzers/libFuzzer/libfuzzer_launcher_test.py]: This functionality was already broken for 2 minijail tests. since command[0] is minijail bin.
